keyboards/hardwareabstraction/handwire/keymaps/default/keymap.c

```diff
@@ -0,0 +1,315 @@
+        KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,     KC_TRNS,    KC_TRNS,      KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS,                           KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS
+    )
+};
+
+// Documentation: custom_quantum_functions.md
+void keyboard_post_init_user(void){
+    setPinOutput(BUZZER_PIN);
+}
+
+void housekeeping_task_user(void){
+    if(buzzer_on){
+        if(buzzer_active && timer_elapsed(buzzer_timer) > buzzer_dwell){
+        buzzer_active = false;
+        writePinLow(BUZZER_PIN);
+        }
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed){                         // true when pressed, false when released
+        initial_keypress = true;
+
+        if(buzzer_on){
+            if(!buzzer_active){
+                buzzer_active = true;
+                buzzer_timer = timer_read();
+                writePinHigh(BUZZER_PIN);
+            }
+        }
+
+        switch (keycode) {
+            #ifdef HAPTIC_ENABLE
+            case KC_HPTON:
+                haptic_enable();
+                break;
+
+            case KC_HPTOFF:
+                haptic_disable();
+                break;
+
+            case KC_HPTTOG:
+                haptic_toggle();
+                break;
+
+            case KC_HPTRST:
+                haptic_reset();
+                break;
+
+            case KC_HPTFBK:
+                haptic_feedback_toggle();
+                break;
+
+            case KC_HPTCONT:
+                haptic_toggle_continuous();
+                break;
+
+            case KC_HPTDWLI:
+                haptic_dwell_increase();
+                break;
+
+            case KC_HPTDWLD:
+                haptic_dwell_decrease();
+                break;
+            #endif
+
+            case KC_BZTOG:
+                buzzer_on = !buzzer_on;
+                if(buzzer_on == true){
+                    buzzer_active = true;
+                    buzzer_timer = timer_read();
+                    writePinHigh(BUZZER_PIN);
+                }
+                else{
+                    writePinLow(BUZZER_PIN);
+                }
+
+                break;
+
+            case KC_BZRST:
+                buzzer_dwell = 12;
+                break;
+
+            case KC_BZDWLI:
+                buzzer_dwell += buzzer_dwell_change;
+                break;
+
+            case KC_BZDWLD:
+                buzzer_dwell -= buzzer_dwell_change;
+                break;
+        }
+    }
+    return true;
+}
+
+
+#ifdef OLED_ENABLE
```

Comment:
Ideally, this oled code should go into the keyboard's c file (not present), and moved out of the keymaps.  You can see an example of how to handle this in a number of boards, such as the boardsource lulu.

This is because we want to keep the default keymap as simple as possible, for newer users.  (additionally, any behavior defined in the keymap is not present when using the configurator, but kb level stuff will still be)

Reply:
Changed with commit: [e165596](https://github.com/qmk/qmk_firmware/pull/18504/commits/e16559612edd150447a2e450b97677e897f4a961)